src/client/signing.rs

```diff
@@ -105,3 +57,23 @@ where
         }
     }
 }
+
+/// Computes SHA-256 hash of inner object
+pub fn hash_signed_object<T>(trussed: &mut T, data: &[u8]) -> Result<trussed::types::ShortData, ()>
+where
+    T: trussed::client::CryptoClient,
+{
+    let signed_object = trussed::cbor_deserialize::<SignedObject>(data).map_err(|e| {
+        error!("Failed to deserialize signed object (outer): {}", e);
+    })?;
+
+    let sha = trussed::try_syscall!(trussed.hash(
```

Comment:
Is this the only place where `MAX_MESSAGE_LENGTH` limits us?

It is sad that Trussed still uses `update` under the hood (https://github.com/trussed-dev/trussed/blob/main/src/mechanisms/sha256.rs#L43) but doesn't expose it. It should be possible to implement in a stateless way, by passing internal state (same size as hash, plus maybe additional integer for keeping size info) for each function...

Reply:
@krystian-hebel It limits file size too

Reply:
Increased limit to 8192 and tested whether still works on nRF.